core/src/main/java/com/linecorp/armeria/client/HttpClientFactory.java

```diff
@@ -381,6 +382,16 @@ public int numConnections() {
         return pools.values().stream().mapToInt(HttpChannelPool::numConnections).sum();
     }
 
+    /**
+     * Registers a JVM shutdown hook that closes this {@link HttpClientFactory} when the current JVM terminates.
+     *
+     * @param whenClosing the {@link Runnable} will be run before closing this {@link HttpClientFactory}
+     */
```

Comment:
We can probably remove this Javadoc because it's not very different from that of its supertype.